app/src/main/java/com/example/erabook/fragments/discoverGoogle/CameraFragment.kt

```diff
@@ -31,135 +31,135 @@ import com.google.mlkit.vision.common.InputImage
+      }
     }
-
-    private fun startCamera() {
-        val cameraProviderFuture = ProcessCameraProvider.getInstance(requireContext())
-        cameraProviderFuture.addListener({
-            val cameraProvider: ProcessCameraProvider = cameraProviderFuture.get()
-            val preview = Preview.Builder()
-                .build()
-                .also {
-                    it.setSurfaceProvider(binding.viewFinder.surfaceProvider)
-                }
-            imageCapture = ImageCapture.Builder()
-                .build()
-
-            try {
-                cameraProvider.unbindAll()
-                cameraProvider.bindToLifecycle(
-                    viewLifecycleOwner, CameraSelector.DEFAULT_BACK_CAMERA, preview, imageCapture
-                )
-            } catch (e: Exception) {
-                Log.d("Binding error", "Use case binding failed: $e")
-            }
-        }, ContextCompat.getMainExecutor(requireContext()))
-
-    }
-
-    private fun requestCameraPermissions() {
-        activityResultLauncher.launch(REQUESTED_PERMISSIONS)
-    }
-
-    private fun takePhoto() {
-        imageCapture?.takePicture(
-            ContextCompat.getMainExecutor(requireContext()),
-            object : ImageCapture.OnImageCapturedCallback() {
-                override fun onCaptureSuccess(image: ImageProxy) {
-                    val bitmap = image.convertImageProxyToBitmap()
-                    image.close()
-                    processImage(bitmap)
-                }
-
-                override fun onError(exception: ImageCaptureException) {
-                    requireContext().showToast(R.string.default_error)
-                    Log.d("Error capture", "Photo capture failed: ${exception.message}")
-                }
+  }
+
+  private fun allPermissionsGranted() =
+      REQUESTED_PERMISSIONS.all {
+        ContextCompat.checkSelfPermission(requireContext(), it) == PackageManager.PERMISSION_GRANTED
+      }
+
+  private fun startCamera() {
+    val cameraProviderFuture = ProcessCameraProvider.getInstance(requireContext())
+    cameraProviderFuture.addListener(
+        {
+          val cameraProvider: ProcessCameraProvider = cameraProviderFuture.get()
+          val preview =
+              Preview.Builder().build().also {
+                it.setSurfaceProvider(binding.viewFinder.surfaceProvider)
+              }
+          imageCapture = ImageCapture.Builder().build()
+
+          try {
+            cameraProvider.unbindAll()
+            cameraProvider.bindToLifecycle(
+                viewLifecycleOwner, CameraSelector.DEFAULT_BACK_CAMERA, preview, imageCapture)
+          } catch (e: Exception) {
+            Log.d("Binding error", "Use case binding failed: $e")
+          }
+        },
+        ContextCompat.getMainExecutor(requireContext()))
+  }
+
+  private fun requestCameraPermissions() {
+    activityResultLauncher.launch(REQUESTED_PERMISSIONS)
+  }
+
+  private fun takePhoto() {
+    imageCapture?.takePicture(
```

Comment:
What if imageCapture is null, you're not handling that. User should get a proper info from the app when something goes wrong. Otherwise user will keep waiting without any info shown or resulting with uninstalling the app.